ext/spl/spl_fixedarray.c

```diff
@@ -389,10 +389,14 @@ static inline void spl_fixedarray_object_write_dimension_helper(spl_fixedarray_o
 		zend_throw_exception(spl_ce_RuntimeException, "Index invalid or out of range", 0);
 		return;
 	} else {
-		if (!Z_ISUNDEF(intern->array.elements[index])) {
-			zval_ptr_dtor(&(intern->array.elements[index]));
+		/* Fix #81429 */
+		zval *const ptr = &(intern->array.elements[index]);
+		zval tmp;
+		ZVAL_COPY_VALUE(&tmp, ptr);
+		ZVAL_COPY_DEREF(ptr, value);
+		if (!Z_ISUNDEF(tmp)) {
```

Comment:
Drop the unnecessary UNDEF check.

Reply:
Done, I was worried there was some unexpected reason for the original code I hadn't seen that could only be fixed in 8.0. Tests still pass.